tests/containers/testenv_prepare.pm

```diff
@@ -0,0 +1,70 @@
+# SUSE's openQA tests
+#
+# Copyright 2009-2013 Bernhard M. Wiedemann
+# Copyright 2013-2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: podman
+# Summary: Running Podman nginx container under systemd
+# Create my_nginx.container file
+# Inform systemd about the new unit file and this creates a nginx.service
+# Start the created nginx service.
+# Verify the status of the nginx service.
+# Publish the port the container is running on.
+# Fetch a predefined website.
+# Maintainer: QE-C team <qa-c@suse.de>
+
+use Mojo::Base 'containers::basetest';
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use utils;
+use containers::common;
+use containers::utils;
+use containers::container_images;
+
+my $systemd_nginx_unit = <<_EOF_;
+[Unit]
+Description=nginx container quadlet file
+
+[Container]
+Image=registry.opensuse.org/opensuse/nginx:latest
+ContainerName=nginx-Quadlet
+PublishPort=80:80
+Volume=/home/nginx/nginx.conf:/etc/nginx/nginx.conf:rw,Z
+Volume=/home/nginx/index.html:/usr/share/nginx/html/index.html:rw,Z
+
+[Install]
+WantedBy=multi-user.target
+_EOF_
+
+sub run {
+    my ($self) = @_;
+    my $unit_name = 'my_nginx';
+
+    select_serial_terminal;
+    my $quadlet = script_output "rpm -ql podman | grep podman/quadlet";
+
+    assert_script_run(' curl --create-dirs -vo /home/nginx/nginx.conf ' . data_url('containers/nginx/') . 'nginx.conf');
+    assert_script_run(' curl --create-dirs -vo /home/nginx/index.html ' . data_url('containers/nginx/') . 'index.html');
+
+    my $podman = $self->containers_factory('podman');
+    $self->{podman} = $podman;
+
+    # create files for generator
+    assert_script_run("podman --version");
+    assert_script_run("$quadlet -version");
+    assert_script_run("echo '$_' >> /etc/containers/systemd/$unit_name.container") foreach (split /\n/, $systemd_nginx_unit);
```

Comment:
If possible, I'd avoid hardcoding a configuration file into an source code,
like that, also needing then a loop to load each line.
The task could be easier moving the `systemd_nginx_unit` content into a separated file, to be then commissioned i.e. via `scp()`.